test/distributed/fsdp/test_fsdp_tp_integration.py

```diff
@@ -0,0 +1,327 @@
+# Owner(s): ["oncall: distributed"]
```

Comment:
Either we do not land this PR with this test file included, or I need to clean it up a bit. This is directly taken from the previous TP + FSDP draft PR with some `TensorFlattener` logic added. I used it to verify locally that the `TensorFlattener` implementation at least works for the previous tests.

In my understanding, the tests in this file will not run in normal CI since the test machines have <= 2 GPUs.